packages/material-ui/src/MenuList/MenuList.js

```diff
@@ -151,6 +152,10 @@ const MenuList = React.forwardRef(function MenuList(props, ref) {
      */
     const currentFocus = ownerDocument(list).activeElement;
 
+    const capturedKeys = ['ArrowDown', 'ArrowUp', 'Home', 'End'];
+
+    if (isSubMenu && capturedKeys.includes(key)) event.stopPropagation();
```

Comment:
Could you let this bubble up and just don't handle the keydown event if `event.defaultPrevented`?

Reply:
I have to capture keyboard events in subMenus or they propagate across all subMenus as they bubble up the stack.

Reply:
Addressed this and pushed.